php/commands/core.php

```diff
@@ -134,7 +134,8 @@ public function download( $args, $assoc_args ) {
 		$locale = \WP_CLI\Utils\get_flag_value( $assoc_args, 'locale', 'en_US' );
 
 		if ( isset( $assoc_args['version'] ) ) {
-			$version = $assoc_args['version'];
+			$version = strtolower( $assoc_args['version'] );
+			$version = ( 'nightly' === $version ? 'trunk' : $version );
 			$download_url = $this->get_download_url($version, $locale, 'tar.gz');
```

Comment:
> Given it's a private method, so we don't need to worry about backwards compat concerns, could we just remove the third argument and make sure the calling function handles the archive file correctly, regardless of type?

That would be fine for the download command - but the update command passes responsibility to WordPress' `Core_Upgrader`, which I believe only expects a `zip`. 

So I think we should either note in the docbloc that `get_download_url()` always returns a zip for trunk version or call `WP_CLI::error()`. The second option doesn't feel necessary.